SPEC.md

```diff
@@ -334,32 +365,202 @@ interpreted as described in the following table:
+
+Note that only literal whitespace is escaped; whitespace escapes (`\n` and
+such) are retained. For example, these strings are all semantically identical:
+
+```kdl
+"Hello\       \nWorld"
+
+    "Hello\n\
+    World"
+
+"Hello\nWorld"
+
+"
+  Hello
+  World
+  "
+```
+
+##### Invalid escapes
+
+Except as described in the escapes table, above, `\` *MUST NOT* precede any
+other characters in a string.
 
 ### Raw String
 
 Raw Strings in KDL are much like [Quoted Strings](#quoted-string), except they
 do not support `\`-escapes. They otherwise share the same properties as far as
-literal [Newline](#newline) characters go, and the requirement of UTF-8
-representation.
+literal [Newline](#newline) characters go, multi-line rules, and the requirement
+of UTF-8 representation.
 
-Raw String literals are represented as `r`, followed by zero or more `#`
-characters, followed by `"`, followed by any number of UTF-8 literals. The
-string is then closed by a `"` followed by a _matching_ number of `#`
-characters. This allows them to contain raw `"` or `#` characters; only the
-precise terminator (resembling `"##`, for example) ends the raw string. This
-means that the string sequence `"` or `"#` and such must not match the closing
-`"` with the same or more `#` characters as the opening `r`.
+Raw String literals are represented with one or more `#` characters, followed
+by `"`, followed by any number of UTF-8 literals. The string is then closed by
+a `"` followed by a _matching_ number of `#` characters. This means that the
+string sequence `"` or `"#` and such must not match the closing `"` with the
+same or more `#` characters as the opening `#`, in the body of the string.
+
+Like other Strings, Raw Strings _MUST NOT_ include any of the [disallowed
+literal code-points](#disallowed-literal-code-points) as code points in their
+body. Unlike with Quoted Strings, these cannot simply be escaped, and are thus
+unrepresentable when using Raw Strings.
 
 #### Example
 
 ```kdl
-just-escapes r"\n will be literal"
-quotes-and-escapes r#"hello\n\r\asd"world"#
+just-escapes #"\n will be literal"#
+```
+
+The string contains the literal characters `\n will be literal`.
+
+```kdl
+quotes-and-escapes ##"hello\n\r\asd"#world"##
+```
+
+The string contains the literal characters `hello\n\r\asd"#world`
+
+
+### Multi-line Strings
+
+When a Quoted or Raw String spans multiple lines with literal, non-escaped
+Newlines, it follows a special multi-line syntax that automatically "dedents"
+the string, allowing its value to be indented to a visually matching level if
+desired.
+
+A Multi-line string _MUST_ start with a [Newline](#newline) immediately
+following its opening `"`. Its final line _MUST_ contain only whitespace,
+followed by a single closing `"`. All in-between lines that contain
+non-whitespace characters _MUST_ start with the exact same whitespace as the
+final line (precisely matching codepoints, not merely counting characters).
```

Comment:
Just to be clear, this means that the string *must* include a trailing empty line with whitspace? These are all illegal:

```
     "
    Hello
    World"

    "
    Hello
    World
"
```

Reply:
According to the grammar (and #320), the first one should be illegal, the second one should be fine.